ext/strings.go

```diff
@@ -44,6 +57,46 @@ import (
 //	'hello'.charAt(5)  // return ''
 //	'hello'.charAt(-1) // error
 //
+// # Format
+//
+// Returns a new string with substitutions being performed, printf-style.
+// The valid formatting clauses are:
+//
+// `%s` - substitutes a string. This can also be used on lists, maps, bytes, Duration and Timestamp.
+//	Note that the dot/period decimal separator will always be used when printing a list or map that contains a double.
+// `%d` - substitutes an integer.
+// `%f` - substitutes a double with fixed-point precision. The default precision is 6, but
+// this can be adjusted. The strings `Infinity`, `-Infinity`, and `NaN` are also valid input
+// for this clause.
+// `%e` - substitutes a double in scientific notation. The default precision is 6, but this
+// can be adjusted.
+// `%E` - same as a above, but with an uppercase E.
+// `%b` - substitutes an integer with its equivalent binary string.
+// `%x` - substitutes an integer with its equivalent in hexadecimal, or if given a string or
+// bytes, will output each character's equivalent in hexadecimal.
+// `%x` - same as above, but with A-F capitalized.
+// `%o` - substitutes an integer with its equivalent in octal.
+// `%t` - substitutes a boolean with the strings "true" or "false".
+//
+// <string>.format(<list>)` -> <string>
```

Comment:
Is it possible to print a type? (e.g. `%T` in Go)

Have we made a clear decision not to print objects? Can we add test case showing that case is an error with `%s` if we are not supporting it?

cc @TristonianJones

Reply:
`string(type(<value>))` SGTM

I think I'm OK without any object print support. I agree we can add in utilities to k8s if we want that. A test that verifies that attempting to print an object results in a sane error (and not surprising behavior) would be nice.

Reply:
Right, I've added a test that shows printing an arbitrary object is unsupported.